### PR TITLE
feat: add bootcamp cost calc guide

### PR DESCRIPTION
**Pull request by ronaldcs**
- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [X] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

**Comment by ronaldcs** [comment]
@ValeraS here's the bootcamp cost calc component redone based on your component example (ref: https://github.com/freeCodeCamp/freeCodeCamp/pull/34953#issuecomment-463562257). Not familiar with graphql, so let me know if I did it wrong or if there's a better way.

**Comment by RandellDawson** [comment]
Shouldn't this have been added to the original PR?

**Comment by ronaldcs** [comment]
> Shouldn't this have been added to the original PR?

@RandellDawson the original PR had proof-of-concept code to have the ability to place custom components in guide articles, which ValeraS already implemented and merged in a separate PR - https://github.com/freeCodeCamp/freeCodeCamp/pull/35053. I felt we should close the old PR since it's not needed any more and created a new PR to use ValeraS' implementation to add just the bootcamp cost calc portion. Let me know if I did this wrong.

**Comment by RandellDawson** [comment]
@ronaldcs So should we close https://github.com/freeCodeCamp/freeCodeCamp/pull/34953 or am I still missing something?

**Comment by ronaldcs** [comment]
@RandellDawson yes, I believe we should...I can go ahead and do it, but wasn't sure if there was something more formal.

**Comment by RandellDawson** [comment]
@ronaldcs You can add a comment when you close it, that another PR (this one) replaces it.

**Comment by ValeraS** [comment]
@ronaldcs Could you try how it would be if you put all the static markup in the MD file and embed the components using the portals? The main idea is to be able to translate such a component as other articles in the Guide.

**Comment by ronaldcs** [comment]
Understood, I'll take a stab at that.

**Comment by ronaldcs** [comment]
@ValeraS was this something you were looking for?

File: client/src/templates/Guide/components/calculators/CodingBootcampCostCalculator/CodingBootcampCostCalculator.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    incomes: [],
+    city: null,
+    cityLabel: '_______',
+    lastYearsIncome: null,
+    lastYearsIncomeLabel: '_______',
+    bootcamps: null
+  };
+
+  initComponent() {
+    fetch('/json/bootcamps.json')
```
Comment (ValeraS) [inline client/src/templates/Guide/components/calculators/CodingBootcampCostCalculator/CodingBootcampCostCalculator.js]: Let's place `bootcamps.json` in to `md` file, like this:
```html
<script type="application/json" id="bootcamps">
[
  {
    "name": "Hack Reactor",
    "cost": "17780",
    "housing": "500",
    "weeks": "12",
    "finance": true,
    "cities": ["new-york-city", "san-francisco"]
  },
...
</script>
```
and replace link to `bootcamps.json`:
```html
<a
  href='javascript:void(0)'
  download='bootcamps.json'
  id='bootcamps-data-link'
>
  Save Data Source JSON
</a>
```
then in component, we can do somethig like this:
```js
    const bootcamps = JSON.parse(
      document.getElementById('bootcamps').innerHTML
    );
    document
      .getElementById('bootcamps-data-link')
      .setAttribute(
        'href',
        `data:text/plain;charset=utf-8,${encodeURIComponent(
          JSON.stringify(bootcamps, null, 2)
        )}`
      );
```

**Review by ValeraS — CHANGES_REQUESTED** [review]
Sorry for the long delay. I made a few suggestions. If you agree let's do it.

**Comment by ronaldcs** [comment]
@ValeraS sounds GTM, I've added your requested changes.

**Comment by RandellDawson** [comment]
@QuincyLarson What do you want to do about this PR?

**Comment by QuincyLarson** [comment]
@ronaldcs thanks for your extreme patience and for your contribution. We are planning to sunset the guide soon, so this won't actually make it into production. But we will use your code as a reference when we re-instantiate this coding bootcamp cost calculator as a static page later. I am going to go ahead and merge your PR so you get the credit you deserve for your contribution.

**Comment by raisedadead** [comment]
@QuincyLarson I am reverting this PR. This is going to break our builds.

**Pull request merged**